<!-- sct-test-summary -->
## :x: Test Summary: FAILED

### :white_check_mark: Precommit: PASSED

| Total | Passed | Failed | Skipped |
|-------|--------|--------|---------|
| 122 | 23 | 0 | 99 |


### :x: Tests: FAILED

| Total | Passed | Failed | Errors | Skipped |
|-------|--------|--------|--------|---------|
| 3591 | 3552 | 1 | 0 | 38 |

### Failed Tests

#### `integration.test_config.test_18_error_if_no_version_repo_ami_selected`
**Type:** FAILURE
**Message:** `AssertionError: Regex pattern did not match.
  Expected regex: "scylla version/repos wasn't configured correctly"
  Actual message: 's3_baremetal_config missing from config for baremetal'`
<details><summary>Traceback</summary>

```
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x74823a3ae780>

    @pytest.mark.integration
    def test_18_error_if_no_version_repo_ami_selected(monkeypatch):
        monkeypatch.delenv("SCT_AMI_ID_DB_SCYLLA", raising=False)
        monkeypatch.delenv("SCT_SCYLLA_VERSION", raising=False)
        for backend in sct_config.available_backends:
            if "k8s" in backend:
                continue
            if "siren" in backend:
                continue
            if backend == "xcloud":
                monkeypatch.setenv("SCT_XCLOUD_PROVIDER", "aws")
            monkeypatch.setenv("SCT_CLUSTER_BACKEND", backend)
            monkeypatch.setenv("SCT_CONFIG_FILES", "unit_tests/test_configs/minimal_test_case.yaml")
    
            conf = sct_config.SCTConfiguration()
            with pytest.raises(AssertionError, match="scylla version/repos wasn't configured correctly") as context:
>               conf.verify_configuration()

unit_tests/integration/test_config.py:279: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sdcm/sct_config.py:3447: in verify_configuration
    self._check_per_backend_required_values(backend)
sdcm/sct_config.py:3792: in _check_per_backend_required_values
    self._check_backend_defaults(backend, self.backend_required_params[backend])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = SCTConfiguration(perf_simple_query_extra_command=None, force_run_iotune=None, data_volume_disk_throughput=None, multi_...d_regions=['eu-west-1', 'eu-west-2', 'eu-west-3', 'us-west-2', 'us-east-1', 'us-east-2', 'eu-north-1', 'eu-central-1'])
backend = 'baremetal'
required_params = ['s3_baremetal_config', 'user_credentials_path']

    def _check_backend_defaults(self, backend, required_params):
        fields = [
            field_name
            for field_name, field in self.__class__.model_fields.items()
            if field_name in required_params and not is_ignored_field(field
```
</details>

[Full build log](https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-14955/7/)

--- a/sdcm/sct_events/events_device.py
+++ b/sdcm/sct_events/events_device.py
@@ -222,7 +222,7 @@ class EventsDevice(multiprocessing.Process):
                 yield obj.base, obj
 
     def is_alive(self) -> bool:
-        return self._running.is_set()
+        return self._running.is_set() and super().is_alive()
 
 
 start_events_main_device = partial(start_events_process, EVENTS_MAIN_DEVICE_ID, EventsDevice)

--- a/sdcm/sct_events/events_device.py
+++ b/sdcm/sct_events/events_device.py
@@ -91,6 +91,8 @@ class EventsDevice(multiprocessing.Process):
     def stop(self, timeout: Optional[float] = None) -> None:
         self._running.clear()
         self.join(timeout)
+        # self.is_alive() returns self._running.is_set(), which is already False here
+        # (cleared above). Call super() to check the actual OS-level process liveness.
         if super().is_alive():
             LOGGER.warning("EventsDevice did not stop within timeout, killing")
             self.kill()

--- a/sdcm/sct_events/events_device.py
+++ b/sdcm/sct_events/events_device.py
@@ -15,6 +15,7 @@ import time
 import queue
 import ctypes
 import pickle
+import signal
 import logging
 import multiprocessing
 from typing import Optional, Generator, Any, Tuple, Callable, cast, Dict
@@ -91,10 +92,12 @@ class EventsDevice(multiprocessing.Process):
     def stop(self, timeout: Optional[float] = None) -> None:
         self._running.clear()
         self.join(timeout)
-        # self.is_alive() returns self._running.is_set(), which is already False here
-        # (cleared above). Call super() to check the actual OS-level process liveness.
         if super().is_alive():
-            LOGGER.warning("EventsDevice did not stop within timeout, killing")
+            LOGGER.warning("EventsDevice did not stop within timeout, sending SIGTERM")
+            self.terminate()
+            self.join(5)
+        if super().is_alive():
+            LOGGER.warning("EventsDevice still alive after SIGTERM, sending SIGKILL")
             self.kill()
             self.join(5)
         self._queue.cancel_join_thread()
@@ -107,7 +110,10 @@ class EventsDevice(multiprocessing.Process):
         raise RuntimeError("EventsDevice is not ready to send events.")
 
     def run(self):
-        with suppress_interrupt(), verbose_suppress("EventsDevice failed"):
+        # SIGTERM handler enables cooperative shutdown: terminate() sends SIGTERM which
+        # triggers _running.clear(), letting the loop drain and exit gracefully.
+        signal.signal(signal.SIGTERM, lambda *_: self._running.clear())
+        with verbose_suppress("EventsDevice failed"):
             with zmq.Context() as ctx, ctx.socket(zmq.PUB) as pub, ctx.socket(zmq.SUB) as sub:
                 self._sub_port.value = pub.bind_to_random_port("tcp://*")
                 self._running.set()
@@ -222,7 +228,14 @@ class EventsDevice(multiprocessing.Process):
                 yield obj.base, obj
 
     def is_alive(self) -> bool:
-        return self._running.is_set() and super().is_alive()
+        if not self._running.is_set():
+            return False
+        try:
+            return super().is_alive()
+        except AssertionError:
+            # CPython's BaseProcess.is_alive() asserts self._parent_pid == os.getpid().
+            # If called from a forked child, fall back to the shared event flag.
+            return True
 
 
 start_events_main_device = partial(start_events_process, EVENTS_MAIN_DEVICE_ID, EventsDevice)

--- a/sdcm/sct_events/events_processes.py
+++ b/sdcm/sct_events/events_processes.py
@@ -101,7 +101,7 @@ class BaseEventsProcess(Generic[T_inbound_event, T_outbound_event], abc.ABC):
         self.join(timeout)
         if self.is_alive():
             LOGGER.error("Events process %s is still alive after %s timeout", events_process_class_name, timeout)
-            if hasattr(self, "kill"):
+            if hasattr(self, "kill"):  # threading.Thread has no kill(); only escalate for Process-backed subclasses
                 self.kill()
                 self.join(2)
         else:

--- a/sdcm/sct_events/events_processes.py
+++ b/sdcm/sct_events/events_processes.py
@@ -101,7 +101,7 @@ class BaseEventsProcess(Generic[T_inbound_event, T_outbound_event], abc.ABC):
         self.join(timeout)
         if self.is_alive():
             LOGGER.error("Events process %s is still alive after %s timeout", events_process_class_name, timeout)
-            if hasattr(self, "kill"):  # threading.Thread has no kill(); only escalate for Process-backed subclasses
+            if isinstance(self, multiprocessing.Process):
                 self.kill()
                 self.join(2)
         else:

--- a/sdcm/tester.py
+++ b/sdcm/tester.py
@@ -3785,9 +3785,9 @@ class ClusterTester(unittest.TestCase):
 
     @silence()
     def clean_resources(self):
+        self._stop_all_node_task_threads()
         if not self.params.get("execute_post_behavior"):
             self.log.info("Resources will continue to run")
-            self._stop_all_node_task_threads()
             return
 
         actions_per_cluster_type = get_post_behavior_actions(self.params)

